distributed/scheduler.py

```diff
@@ -1310,17 +1310,37 @@ class TaskState:
     #: Task annotations
     annotations: dict[str, Any]
 
+    #: The unique identifier of a specific execution of a task. This identifier
+    #: is used to sign a task such that the assigned worker is expected to return
+    #: the same identifier in the task-finished message. This is used to correlate
+    #: responses.
+    #: Only the most recently assigned worker is trusted. All other results will
+    #: be rejected.
+    run_id: int
+
+    #: Whether the task has already been attempted. This value is used to refresh
+    #: refresh the run ID on following attempts.
+    _attempted: bool
```

Comment:
nit: I consider this an unnecessary complexity since we don't care about the absolute value of the ID but rather about whether or not it changed between submission and result.
It's small and contained so that's not a big deal